dataset/src/main/scala/frameless/functions/AggregateFunctions.scala

```diff
@@ -7,12 +7,15 @@ import org.apache.spark.sql.{functions => untyped}
 
 trait AggregateFunctions {
 
+  private def typedColumnToAggregate[A: TypedEncoder, T](a: TypedColumn[T, A]): TypedAggregate[T, A] =
+    new TypedAggregate[T,A](a.expr)
+
   /** Creates a [[frameless.TypedColumn]] of literal value. If A is to be encoded using an Injection make
     * sure the injection instance is in scope.
     *
     * apache/spark
     */
```

Comment:
@OlivierBlanvillain, in Scala you cannot have two overloaded functions with same parameters but different return types, right? A => B and A => C